src/Maestro/tests/Maestro.ScenarioTests/ObjectHelpers/BuildRefComprarer.cs

```diff
@@ -0,0 +1,20 @@
+using System.Collections.Generic;
+using Microsoft.DotNet.Maestro.Client.Models;
+
+namespace Maestro.ScenarioTests.ObjectHelpers
+{
+    public class BuildRefComprarer : IEqualityComparer<BuildRef>
+    {
+        public bool Equals(BuildRef x, BuildRef y)
+        {
+            return x.BuildId == y.BuildId &&
+                x.IsProduct == y.IsProduct &&
+                x.TimeToInclusionInMinutes == y.TimeToInclusionInMinutes;
+        }
+
+        public int GetHashCode(BuildRef obj)
+        {
+            return obj.GetHashCode();
```

Comment:
Is this a valid implementation?  Does BuildRef define GetHashCode?  I think this should probably be `HashCode.Combine(obj.BuildId, obj.IsProduct, obj.TimeToInclusionInMinutes)`

Reply:
I think it calls the base/default version, but it's certainly easy enough to just implement it.

Reply:
The problem is that Object.GetHashCode() isn't correct once you've overridden Equals, because it will return different answers for two things, even though they are "Equal".

They always come in a matched set (which sort of implies there should be some nice built in way to say "these are the fields that are identifiers, do both methods right".

Reply:
That makes a lot of sense, I hadn't thought about it that way.